core/src/main/java/org/elasticsearch/common/network/Cidrs.java

```diff
@@ -113,4 +113,9 @@ static String octetsToCIDR(int[] octets, int networkMask) {
         assert octets.length == 4;
         return octetsToString(octets) + "/" + networkMask;
     }
+
+    public static String createCIDR(long ipAddress, int networkMask) {
+        long blockSize = 1L << (32 - networkMask);
+        return octetsToCIDR(longToOctets(ipAddress - (ipAddress & (blockSize - 1))), networkMask);
+    }
```

Comment:
agreed that's what I said I don't have a good alternative so let's keep it. But can we make it less lenient and expect an ip address that is compatible with the mask and move the  ` - (ipAddress & (blockSize - 1))` part to the ipv4 range tests?